Zeeguu Reader/Pods/Zeeguu-API-iOS/ZeeguuAPI/ZeeguuAPI/Article.swift

```diff
@@ -26,19 +26,98 @@
+	/// The title of this article
 	public var title: String
+	/// The url of this article
 	public var url: String
+	/// The publication date of this article
 	public var date: String
+	/// The summary of this article
 	public var summary: String
+	
+	private var imageURL: String?
+	private var image: UIImage?
 	private var contents: String?
+	private var difficulty: ArticleDifficulty?
+	
+	/// Whether the contents of this article have been retrieved yet
+	public var isContentLoaded: Bool {
+		return contents != nil
+	}
+	
+	/// Whether the difficulty of this article has been calculated yet
+	public var isDifficultyLoaded: Bool {
+		return difficulty != nil
+	}
 	
+	/// The description of this `Article` object. The value of this property will be used whenever the system tries to print this `Article` object or when the system tries to convert this `Article` object to a `String`.
 	public var description: String {
 		let str = feed.description.stringByReplacingOccurrencesOfString("\n", withString: "\n\t")
 		return "Article: {\n\tfeed: \"\(str)\",\n\ttitle: \"\(title)\",\n\turl: \"\(url)\",\n\tdate: \"\(date)\",\n\tsummary: \"\(summary)\",\n\tcontents: \"\(contents)\"\n}"
 	}
 	
+	// MARK: Static methods -
+	
+	/// Get difficulty for all given articles
+	///
+	/// Note: This method uses the default difficulty computer.
+	///
+	/// - parameter articles: The articles for which to get difficulties. Please note that all `Article` objects are references and once `completion` is called, the given `Article` objects have difficulties.
+	/// - parameter completion: A block that will indicate success. If `success` is `true`, all `Article` objects have been given their difficulty. Otherwise nothing has happened to the `Article` objects. If `articles` is empty, `success` is `false`.
+	public static func getDifficultiesForArticles(articles: [Article], completion: (success: Bool) -> Void) {
+		self.getContentsForArticles(articles, withDifficulty: true, completion: completion)
+	}
+	
+	/// Get contents for all given articles
+	///
+	/// - parameter articles: The articles for which to get contents. Please note that all `Article` objects are references and once `completion` is called, the given `Article` objects have contents.
+	/// - parameter withDifficulty: Whether to calculate difficulty for the retrieved contents. Setting this to `true` will send the language code of the first article's feed as the language of all contents.
+	/// - parameter completion: A block that will indicate success. If `success` is `true`, all `Article` objects have been given their contents. Otherwise nothing has happened to the `Article` objects. If `articles` is empty, `success` is `false`.
+	public static func getContentsForArticles(articles: [Article], withDifficulty: Bool = false, completion: (success: Bool) -> Void) {
+		if articles.count == 0 {
+			return completion(success: false) // No articles to get content for
+		}
+		let urls = articles.map({ $0.url })
```

Comment:
Yes, I'm also liking it more and more 😊

This is a file from the cocoa pod 😊. Comments about the Zeeguu API should go in the Zeeguu API repository 😜.